proposals/0448-type-variable-scoping.rst

```diff
@@ -414,7 +500,19 @@ Proposed Change Specification
    (Right now, pattern synonyms require all such quantifications to occur before any term arguments,
    but accepted proposal `#402`_ allows these quantifications to occur in any order in data constructors.)
 
-#. Any type variables mentioned in a type application are considered binding sites, shadowing any in-scope type variables.
+#. In accordance with the `Visibility Orthogonality Principle`_,
+   the rules that determine whether a variable occurrence is a binding site or a use site are not affected by the presence of a ``@``.
+   At the time of writing, the rules are as follows:
+
+   * Outside pattern signatures,
+     variable occurrences are considered binding sites,
+     shadowing any other in-scope variables.
+     It is an error to bring the same type variable into scope in two (or more) places within the same match group.
+
+   * Inside pattern signatures (i.e. on the right-hand side of ``pat :: sig``),
+     occurrences of in-scope type variables are usages,
+     whereas occurrences of out-of-scope type variables create implicit bindings.
+     It is allowed to mention the same out-of-scope variable more than once.
```

Comment:
This is unrelated to the current principal, isn't it?

Reply:
Oh whoops, I accidentally pushed my merging of https://github.com/ghc-proposals/ghc-proposals/pull/556 into this to fix comments.

Reply:
Forced pushed to remove merge.